### PR TITLE
Implement marker clustering

### PR DESCRIPTION
**Pull request by danwelch3**
Implements M Kergall's [osmbonuspack](https://github.com/MKergall/osmbonuspack) Marker Clustering feature.

I was not able to implement the library as a dependency, as I could not customize the markers to match the markers currently used. I put the relevant parts of the library in with modifications to the markers.

image

### CODE REVIEW

**Comment by jamesarich** [comment]
I'm unsure how copying code directly from other projects needs to be attributed, but it does look like the [license is pretty liberal](https://github.com/MKergall/osmbonuspack#licence).

**Comment by danwelch3** [comment]
Agreed. I would prefer to link directly to the library if possible, so any suggestions on using the markers from com.geeksville.mesh.model.map.MarkerWithLabel instead of the default OSMDroid markers would be appreciated.

Its my impression that in terms of license, this is not a problem.  The osmbonuspack library is licensed under LGPLv3 which should be compatible with GPLv3.  [See GNU's license matrix here.](https://www.gnu.org/licenses/gpl-faq.html#AllCompatibility)

image

File: app/src/main/java/com/geeksville/mesh/ui/map/MapFragment.kt [hunk 1]
```diff
@@ -283,6 +290,9 @@ fun MapView(
     val map = rememberMapViewWithLifecycle(context)
     val state by model.mapState.collectAsStateWithLifecycle()
 
+    val nodeClusterer = RadiusMarkerClusterer(context)
+    map.overlays.add(nodeClusterer)
```
Comment (andrekir) [inline app/src/main/java/com/geeksville/mesh/ui/map/MapFragment.kt]: please move this into `drawOverlays()` and should be ready to merge

**Comment by danwelch3** [comment]
Moved `map.overlays.add(nodeClusterer)` to `MapView.drawOverlays()`.

I left the `nodeClusterer` definition where it was, otherwise it would break the `UpdateMarkers()` call.

**Comment by DonRichie** [comment]
This change reduced the usability of my map. I can't see when I had contact with the nodes without zooming in.
If you insist on the feature, please add a button to turn it off.

**Comment by danwelch3** [comment]
I'll look into adding a toggle.  The clustering shouldn't go into effect until nodes are overlapping visually though.  There might be some tuning of the cluster settings that needs done as well.  Can you attach screenshots of the same nodes with and without this feature showing the problem you're having?

**Comment by andrekir** [comment]
there is no need for a toggle. the clustering was fine tuned in app version 2.5.1, please update.

**Pull request merged**